guardian/recipes-backend
# Write article ids for reindex

## What does this change?

Write article ids in our snapshot, rather than the recipe index; it's all we need to provide to the reindex queue.

## Why didn't we do this before?

Before this PR, the assumption was that we would reindex on a per-recipe basis.

However, we fetch recipes from our reindex source, CAPI, on a per-article basis.

It's inefficient, for an article that contains many recipes, to fetch that article many times to retrieve each recipe for a reindex. So we now reindex recipes on a per-article basis, to ensure we only fetch each article once for all the recipes it contains.

## How to test

- Observe that the code and test fixture have been refactored to pull data straight from dynamoDB, and write a list of article paths to S3 (rather than the entire `RecipeIndex` structure.
- Deploy to CODE, and exercise the step function. It should run successfully, and log healthy output:

<img width="1312" alt="Screenshot 2024-12-19 at 15 14 13" src="https://github.com/user-attachments/assets/c1f609cf-9a37-494b-80f7-8978b7e39d84" />
